src/modules/map/components/toolbox/TimeSlider.vue

```diff
@@ -1,3 +1,305 @@
+
+<script setup>
+import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
+import { computed, onMounted, onUnmounted, ref, watch } from 'vue'
+import { useStore } from 'vuex'
+
+import { YEAR_TO_DESCRIBE_ALL_OR_CURRENT_DATA } from '@/api/layers/LayerTimeConfigEntry.class'
+import { round } from '@/utils/numberUtils'
+
+const dispatcher = { dispatcher: 'TimeSlider.vue' }
+
+/**
+ * The oldest year in our system is from the layer Journey Through Time (ch.swisstopo.zeitreihen)
+ * which has data from the year 1844
+ *
+ * @type {Number}
+ */
+const OLDEST_YEAR = 1844
+
+/**
+ * The youngest (closest to now) year in our system, it will always be the previous year as of now
+ *
+ * @type {Number}
+ */
+const YOUNGEST_YEAR = new Date().getFullYear() - 1
+
+const ALL_YEARS = (() => {
+    const years = []
+    for (let year = OLDEST_YEAR; year <= YOUNGEST_YEAR; year++) {
+        years.push(year)
+    }
+    return years
+})()
+
+const LABEL_WIDTH = 32
+const MARGIN_BETWEEN_LABELS = 50
+const PLAY_BUTTON_SIZE = 54
+
+/**
+ * Finds the most recent common year between all given time configs
+ *
+ * @param {LayerTimeConfig[]} timeConfigs
+ * @returns {Number} Most recent common year between all given time configs
+ */
+function findMostRecentCommonYear(timeConfigs) {
+    if (timeConfigs.length < 2) {
+        return null
+    }
+    let yearsInCommon = [...timeConfigs[0].years]
+    timeConfigs.slice(1).forEach((timeConfig) => {
+        yearsInCommon = yearsInCommon.filter((year) => timeConfig.years.includes(year))
+    })
+    return yearsInCommon[0]
+}
+
+// dynamic internal data
+const sliderWidth = ref(0)
+const allYears = ref(ALL_YEARS)
+const currentYear = ref(YOUNGEST_YEAR)
+const cursorX = ref(0)
+const playYearsWithData = ref(false)
+const yearCursorIsGrabbed = ref(false)
+const playYearInterval = ref(null)
```

Comment:
I don't think you should store that in a ref, a simple `const` or `let` would do the trick